ziprofs.py

```diff
@@ -198,37 +194,56 @@ def readdir(self, path, fh):
         result.extend(subdirs)
         return result
 
-
     def release(self, path, fh):
-        zip_path = self.get_zip_path(path)
-        if not zip_path:
-            return os.close(fh)
-        return 0
+        if fh in self._zip_file_fh:
+            with self._lock:
+                f = self._zip_file_fh[fh]
+                with self._zip_zfile_fh[fh].lock():
+                    del self._zip_file_fh[fh]
+                    del self._zip_zfile_fh[fh]
+                    return f.close()
+        else:
+            return os.close(fh >> 1)
 
     def statfs(self, path):
         stv = os.statvfs(path)
-        return dict((key, getattr(stv, key)) for key in ('f_bavail', 'f_bfree',
-            'f_blocks', 'f_bsize', 'f_favail', 'f_ffree', 'f_files', 'f_flag',
-            'f_frsize', 'f_namemax'))
+        return dict((key, getattr(stv, key)) for key in (
+            'f_bavail', 'f_bfree', 'f_blocks', 'f_bsize', 'f_favail',
+            'f_ffree', 'f_files', 'f_flag', 'f_frsize', 'f_namemax'
+        ))
+
+
+def parse_mount_opts(in_str):
+    opts = {}
+    for o in in_str.split(','):
+        if '=' in o:
+            name, val = o.split('=', 1)
+            opts[name] = val
+        else:
+            opts[o] = True
+    return opts
 
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(
         description='ZipROFS read only transparent zip filesystem.',
         formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-    parser.add_argument('ROOT', nargs='?', help="filesystem root")
-    parser.add_argument('MOUNTPOINT', nargs='?', help="filesystem mount point")
+    parser.add_argument('root', nargs='?', help="filesystem root")
+    parser.add_argument('mountpoint', nargs='?', help="filesystem mount point")
     parser.add_argument(
         '-o', metavar='options', dest='opts',
-        help="comma separated list of options: foreground, debug, allowother")
-    args = parser.parse_args()
-    opts = args.opts.split(',') if args.opts else []
-    print(opts)
+        help="comma separated list of options: foreground, debug, allowother, cachesize=N",
+        type=parse_mount_opts, default={})
+    arg = parser.parse_args()
+
+    if 'cachesize' in arg.opts:
+        CachedZipFactory.MAX_CACHE_SIZE = int(arg.opts['cachesize'])
```

Comment:
Check that int is in some reasonable range, like [1, 10M]

Reply:
I'll check if it is greater than 1, I don't feel like checking upper bound is good idea.